test/e2e/util/util.go

```diff
@@ -1683,3 +1683,43 @@ func EnsureAllReqServingPodsLandOnReqServingNodes(t *testing.T, ctx context.Cont
 		g.Expect(node.Labels).To(HaveKeyWithValue(hyperv1.RequestServingComponentLabel, "true"))
 	}
 }
+
+func EnsurePodsDontLandInDefaultNodePool(t *testing.T, ctx context.Context, mgtClient crclient.Client, client crclient.Client, hostedCluster *hyperv1.HostedCluster) {
+	g := NewWithT(t)
+
+	var nodepools hyperv1.NodePoolList
+	if err := mgtClient.List(ctx, &nodepools, crclient.InNamespace(hostedCluster.Namespace)); err != nil {
+		t.Fatalf("failed to list nodepools in guest cluster: %v", err)
+	}
+	g.Expect(nodepools.Items).ToNot(BeEmpty())
+
+	found := false
+	var defaultNodepool hyperv1.NodePool
+	for _, nodepool := range nodepools.Items {
+		if nodepool.Labels["hypershift.openshift.io/auto-created-for-infra"] == hostedCluster.Spec.InfraID {
+			defaultNodepool = nodepool
+			found = true
+			break
+		}
+	}
+
+	if found == false {
+		t.Fatalf("failed to find default nodepool in guest cluster")
+	}
+
+	var defaultNodepoolNodes corev1.NodeList
+	if err := client.List(ctx, &defaultNodepoolNodes, crclient.MatchingLabels{hyperv1.NodePoolLabel: defaultNodepool.Name}); err != nil {
+		t.Fatalf("failed to list nodes in default nodepool: %v", err)
+	}
+	g.Expect(defaultNodepoolNodes.Items).ToNot(BeEmpty())
+
+	for _, node := range defaultNodepoolNodes.Items {
+		var podObj corev1.Pod
+		var podList corev1.PodList
```

Comment:
these Nodes will have pods as part of being operational within its cluster,  the check should validate that none of those are for the HCP, if whitelisting that somehow become too convoluted maybe checking all our HCO pods have the nodeSelector set is good enough

Reply:
should all HCP pods have the `hypershift.openshift.io/hosted-control-plane`  label or what would be the best way to check if a pod is part of HCP/HCO ?